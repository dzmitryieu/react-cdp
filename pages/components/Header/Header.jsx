import React from 'react';
import Link from 'next/link';
import { withRouter } from 'next/router';
import PropTypes from 'prop-types';
import { ESC_KEY } from '../../../src/utils/consts';
import styles from './Header.scss';

class Header extends React.Component {
  onClickHandler = () => {
    const { router } = this.props;
    router.push('/');
  }

  onKeyPressHandler = (e) => {
    if (e.keyCode === ESC_KEY) {
      this.onClickHandler();
    }
  }

  render() {
    return (
      <header className={styles.header}>
        <p
          className={styles.logo}
          onClick={this.onClickHandler}
          onKeyPress={this.onKeyPressHandler}
          role="presentation"
        >
          netflixroulette
        </p>
      </header>
    );
  }
}

Header.propTypes = {
  router: PropTypes.func.isRequired,
};

export default withRouter(Header);
