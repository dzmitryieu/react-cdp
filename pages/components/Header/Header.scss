@import "../../../src/utils/constants.scss";

.header {
  display: flex;
  justify-content: space-between;
  align-items: center;
  z-index: 10;

  a {
    text-decoration: none;
    color: $light-red;
  }

  .logo {
    font-size: 25px;
    letter-spacing: 2px;
    line-height: 32px;
    font-family: $helvetica;
    font-weight: 500;
    cursor: pointer;
  }
}
