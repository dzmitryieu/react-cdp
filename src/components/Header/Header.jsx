import React from 'react';
import PropTypes from 'prop-types';
import { ESC_KEY } from '../../utils/consts';
import styles from './Header.scss';

class Header extends React.Component {
  onClickHandler = () => {
    const { showSearchPage, fetchAllFilms } = this.props;
    showSearchPage();
    fetchAllFilms();
  }

  onKeyPressHandler = (e) => {
    if (e.keyCode === ESC_KEY) {
      this.onClickHandler();
    }
  }

  render() {
    return (
      <header className={styles.header}>
        <p
          className={styles.logo}
          onClick={this.onClickHandler}
          onKeyPress={this.onKeyPressHandler}
          role="presentation"
        >
          netflixroulette
        </p>
      </header>
    );
  }
}


Header.propTypes = {
  showSearchPage: PropTypes.func.isRequired,
  fetchAllFilms: PropTypes.func.isRequired,
};

export default Header;
