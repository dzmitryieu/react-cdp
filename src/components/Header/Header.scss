@import "../../utils/constants.scss";

.header {
  color: $light-red;
  display: flex;
  justify-content: space-between;
  align-items: center;
  z-index: 10;

  .logo {
    font-size: 25px;
    letter-spacing: 2px;
    line-height: 32px;
    font-family: $helvetica;
    font-weight: 500;
    cursor: pointer;
  }
}
