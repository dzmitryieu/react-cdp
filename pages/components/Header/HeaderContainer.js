import { connect } from 'react-redux';
import { withRouter } from 'next/router';
import Header from './Header';
import { fetchAllFilms } from '../../../src/modules/films/filmsActions';
import { showSearchPage } from '../../../src/modules/film/filmActions';

const mapDispatchToProps = {
  fetchAllFilms,
  showSearchPage,
};

export default withRouter(Header);
