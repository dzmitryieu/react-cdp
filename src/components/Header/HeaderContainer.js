import { connect } from 'react-redux';
import Header from './Header';
import {
  fetchAllFilms,
  showSearchPage,
} from '../../modules/films/filmsActions';

const mapDispatchToProps = {
  fetchAllFilms,
  showSearchPage,
};

export default connect(null, mapDispatchToProps)(Header);
