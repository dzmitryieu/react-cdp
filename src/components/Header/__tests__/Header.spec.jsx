import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Header from '../Header';

const renderer = new ShallowRenderer();
Enzyme.configure({ adapter: new Adapter() });

const showSearchPage = jest.fn();
const fetchAllFilms = jest.fn();

describe('Header Snapshot', () => {
  test('renders', () => {
    const component = renderer.render(
      <Header showSearchPage={showSearchPage} fetchAllFilms={fetchAllFilms} />,
    );
    expect(component).toMatchSnapshot();
  });
});

describe('Header clickHandler should works', () => {
  it('click to the logo should call clickHandler', () => {
    const wrapper = shallow(
      <Header showSearchPage={showSearchPage} fetchAllFilms={fetchAllFilms} />,
    );
    const instance = wrapper.instance();
    const spy = jest.spyOn(instance, 'onClickHandler');
    wrapper.instance().forceUpdate();
    wrapper.find('.logo').prop('onClick')();
    expect(spy).toHaveBeenCalled();
  });
  it('keyPress to the esc button on logo should call onKeyPressHandler', () => {
    const wrapper = shallow(
      <Header showSearchPage={showSearchPage} fetchAllFilms={fetchAllFilms} />,
    );
    const e = {
      keyCode: 27,
    };
    const instance = wrapper.instance();
    const spy = jest.spyOn(instance, 'onKeyPressHandler');
    wrapper.instance().forceUpdate();
    wrapper.find('.logo').prop('onKeyPress')(e);
    expect(spy).toHaveBeenCalled();
  });
  it("keyPress not to the esc button on logo shouldn't call onKeyPressHandler", () => {
    const wrapper = shallow(
      <Header showSearchPage={showSearchPage} fetchAllFilms={fetchAllFilms} />,
    );
    const e = {
      keyCode: 0,
    };
    const instance = wrapper.instance();
    const spy = jest.spyOn(instance, 'onKeyPressHandler');
    wrapper.instance().forceUpdate();
    wrapper.find('.logo').prop('onKeyPress')(e);
    expect(spy).toHaveBeenCalled();
  });
});
