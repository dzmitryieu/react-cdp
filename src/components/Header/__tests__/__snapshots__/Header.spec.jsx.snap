// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Snapshot renders 1`] = `
<header
  className="header"
>
  <p
    className="logo"
    onClick={[Function]}
    onKeyPress={[Function]}
    role="presentation"
  >
    netflixroulette
  </p>
</header>
`;

exports[`Header Snapshot renders 2`] = `
<header
  className="header"
>
  <p
    className="logo"
    onClick={[Function]}
    onKeyPress={[Function]}
    role="presentation"
  >
    netflixroulette
  </p>
</header>
`;
